File: backend/app/domains/alerts/schemas/alerts.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class AlertRead(BaseModel):
    id: UUID
    title: str
    alert_type: str
    severity: str
    state: str | None = None
    message: str
    is_active: bool
    created_at: datetime
    expires_at: datetime | None = None


class WatchlistRead(BaseModel):
    id: UUID
    user_id: UUID
    name: str
    watch_type: str
    target_id: str
    is_active: bool
    created_at: datetime | None = None


class BulkUnwatchRequest(BaseModel):
    watchlist_ids: list[UUID]


class WatchlistCreate(BaseModel):
    name: str
    watch_type: str
    target_id: str


class NotificationRead(BaseModel):
    id: UUID
    user_id: UUID
    alert_id: UUID | None = None
    channel: str
    status: str
    read: bool
    created_at: datetime


class NewsArticleRead(BaseModel):
    id: UUID
    title: str
    url: str
    source: str
    category: str
    state: str | None = None
    summary: str | None = None
    image_url: str | None = None
    published_at: datetime | None = None
    scraped_at: datetime
    is_active: bool


class NewsWatchlistCreate(BaseModel):
    article_id: str
    article_title: str

File: frontend/src/types/alerts.ts
export interface Alert {
  id: string;
  title: string;
  alert_type: string;
  severity: string;
  state: string | null;
  message: string;
  is_active: boolean;
  created_at: string;
  expires_at: string | null;
}

export interface Watchlist {
  id: string;
  user_id: string;
  name: string;
  watch_type: string;
  target_id: string;
  is_active: boolean;
}

export interface Notification {
  id: string;
  user_id: string;
  alert_id: string | null;
  channel: string;
  status: string;
  read: boolean;
  created_at: string;
}

export interface NewsArticle {
  id: string;
  title: string;
  url: string;
  source: string;
  category: string;
  state: string | null;
  summary: string | null;
  image_url: string | null;
  published_at: string | null;
  scraped_at: string;
  is_active: boolean;
}

export interface NewsFilters {
  states: string[];
  sources: string[];
}
